### DISCUSSION TITLE
DSP Channel Convertor

### DISCUSSION THREAD

**Discussion by DM1785** [enzo1982/freac#366] [General Discussion]
I just discovered under DSP in General Settings a feature called "Channel Convertor" which has settings for Mono, Stereo, 2.1, 4.0 and 5.1. What does it do, and how does it work? I do have several Surround Sound CDs, both normal CDs and SACDs. I am currently going through my music collection and ripping everything to .flac files for use on the laptop and tablet. My PC has multi-channel output. I have tried searching this forum with nothing found.

![image](https://user-images.githubusercontent.com/111628962/188261697-e9e6cd3c-01fc-4bd3-901b-5027a06b5228.png)

Thank you!

**Comment by enzo1982** [comment]
The _Channel Converter_ component will generally only reduce the number of channels (only exception is going from Mono to Stereo), so it can be used for downmixing 5.1 files to e.g. 2.1 or Stereo, but will not expand your Surround CDs to 5.1.

For that, please use the latest [continuous build](https://github.com/enzo1982/freac/releases/tag/continuous) of fre:ac. It contains a new _Matrix Surround Decoder_ component which will expand Surround CDs to e.g. 5.1 discrete channels.

Note that the DSP components must be enabled on the _Processing_ page in the configuration dialog. Just changing the settings under _Components->DSP_ will not enable them.

**Reply by DM1785** [reply]
# #

> The _Channel Converter_ component will generally only reduce the number of channels (only exception is going from Mono to Stereo), so it can be used for downmixing 5.1 files to e.g. 2.1 or Stereo, but will not expand your Surround CDs to 5.1.
> 
> For that, please use the latest [continuous build](https://github.com/enzo1982/freac/releases/tag/continuous) of fre:ac. It contains a new _Matrix Surround Decoder_ component which will expand Surround CDs to e.g. 5.1 discrete channels.
> 
> Note that the DSP components must be enabled on the _Processing_ page in the configuration dialog. Just changing the settings under _Components->DSP_ will not enable them.

Thank you, @enzo1982. Downmixing is definitely not what I want to do. I have **_A LOT_** of Surround Sound CDs that have the "Intersound" logo, a few with the "Dolby" logo, a few Quadraphonic, and an ever increasing number of SACDs. I want to be able to maintain 4 channel or 5.1 channel format surround sound in the flac files. It will be a while before I can get started on the, because I this I need to buy a Bluray drive for the PC so that it can play enhanced  SACD instead of being stuck with just 2 channel stereo that I currently get when I try to rip an SACD. 

Can the "continuous build" be installed along side the current stable build?

**Reply by enzo1982** [reply]
> Can the "continuous build" be installed along side the current stable build?

Yes, the continuous build is just a ZIP archive. You can extract it anywhere, e.g. to you desktop, and then run it from there.

Regarding SACDs, you probably know that fre:ac does not support SACD ripping. You'll need another tool for that. But you should be able to convert the resulting DSD files with fre:ac if necessary.

For regular Surround CDs, the matrix decoder in the continuous builds can be configured to output different channel configurations. 5.1, i.e. Dolby Pro Logic II, is supported directly and Intersound CDs should be compatible with that. For Quadrophonic select 4.1 and disable the LFE option.
